### ISSUE TITLE
Integration modules import incorrectly

### ISSUE THREAD

**Issue by severn-everett** [Kotlin/kotlinx.coroutines#3089]
Per [this](https://github.com/MicroUtils/kotlin-logging/issues/165) issue, I would like to create a new module that wraps Kotlin-Logging's `withLoggingContext` around a `withContext` block that uses `MDCContext()`. However, there appear to be one or more issues surrounding the import of integration modules. For example, this hypothetical Gradle declaration block:
```
dependencies {
    implementation("io.github.microutils:kotlin-logging-jvm:2.1.16")
    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-guava:1.5.2")
    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-slf4j:1.5.2")
}
```
Attempting to build an integration module with this declaration produces the following warning error:
```
   > Could not find org.jetbrains.kotlinx:kotlinx-coroutines-guava:1.6.0-RC2-SNAPSHOT.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-guava/1.6.0-RC2-SNAPSHOT/maven-metadata.xml
       - https://dl.google.com/dl/android/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-guava/1.6.0-RC2-SNAPSHOT/kotlinx-coroutines-guava-1.6.0-RC2-SNAPSHOT.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-guava/1.6.0-RC2-SNAPSHOT/maven-metadata.xml
       - https://repo.maven.apache.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-guava/1.6.0-RC2-SNAPSHOT/kotlinx-coroutines-guava-1.6.0-RC2-SNAPSHOT.pom
       - https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/org/jetbrains/kotlinx/kotlinx-coroutines-guava/1.6.0-RC2-SNAPSHOT/maven-metadata.xml
       - https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/org/jetbrains/kotlinx/kotlinx-coroutines-guava/1.6.0-RC2-SNAPSHOT/kotlinx-coroutines-guava-1.6.0-RC2-SNAPSHOT.pom
       - https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev/org/jetbrains/kotlinx/kotlinx-coroutines-guava/1.6.0-RC2-SNAPSHOT/maven-metadata.xml
       - https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev/org/jetbrains/kotlinx/kotlinx-coroutines-guava/1.6.0-RC2-SNAPSHOT/kotlinx-coroutines-guava-1.6.0-RC2-SNAPSHOT.pom
     Required by:
         project :kotlinx-coroutines-kotlin-logging > project :kotlinx-coroutines-bom
   > Could not find org.jetbrains.kotlinx:kotlinx-coroutines-slf4j:1.6.0-RC2-SNAPSHOT.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-slf4j/1.6.0-RC2-SNAPSHOT/maven-metadata.xml
       - https://dl.google.com/dl/android/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-slf4j/1.6.0-RC2-SNAPSHOT/kotlinx-coroutines-slf4j-1.6.0-RC2-SNAPSHOT.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-slf4j/1.6.0-RC2-SNAPSHOT/maven-metadata.xml
       - https://repo.maven.apache.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-slf4j/1.6.0-RC2-SNAPSHOT/kotlinx-coroutines-slf4j-1.6.0-RC2-SNAPSHOT.pom
       - https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/org/jetbrains/kotlinx/kotlinx-coroutines-slf4j/1.6.0-RC2-SNAPSHOT/maven-metadata.xml
       - https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/org/jetbrains/kotlinx/kotlinx-coroutines-slf4j/1.6.0-RC2-SNAPSHOT/kotlinx-coroutines-slf4j-1.6.0-RC2-SNAPSHOT.pom
       - https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev/org/jetbrains/kotlinx/kotlinx-coroutines-slf4j/1.6.0-RC2-SNAPSHOT/maven-metadata.xml
       - https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev/org/jetbrains/kotlinx/kotlinx-coroutines-slf4j/1.6.0-RC2-SNAPSHOT/kotlinx-coroutines-slf4j-1.6.0-RC2-SNAPSHOT.pom
     Required by:
         project :kotlinx-coroutines-kotlin-logging > project :kotlinx-coroutines-bom
```

This Gradle dependency declaration works correctly in a POC project. However, attempting to declare the `1.6.0-RC2-SNAPHOT` dependency version for the modules in the POC project produces the same `Could not find org.jetbrains.kotlinx:kotlinx-coroutines-#####:1.6.0-RC2-SNAPSHOT` as above.

In summary:
1. Kotlinx.Coroutines appears to be ignoring integration module version declarations in Gradle automatically pointing to the `1.6.0-RC2-SNAPHOT` version of the integration modules.
2. Gradle cannot find the `1.6.0-RC2-SNAPHOT` version of the integration modules.

**Comment by qwwdfsad** [comment]
Could you please specify how to reproduce this problem? The reproducing project would be especially appreciated.

I've double-checked both published BOM and Gradle metadata and they don't contain any references to any SNAPSHOT versions.

**Comment by qwwdfsad** [comment]
It seems to be a lack of diagnostics in Gradle and its platform API.

It attempts to raise the version of the platform to the current, from `1.5.2` to `1.6.0-RC2-SNAPSHOT`, but as long as slf4 does not depend on ``kotlinx-coroutines-guava` *module*, the resolve is not able to do so.

In general, it's not the best idea to depend on the previous version of the module in the same project, as it easily leads into dependency hell and problems like this.

What you probably need is adding `implementation(project(":kotlinx-coroutines-guava"))`

**Comment by severn-everett** [comment]
Ok, that did it - thanks!
